modules/kmm-creating-moduleloader-image.adoc

```diff
@@ -0,0 +1,140 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_content-type: PROCEDURE
+[id="kmm-creating-moduleloader-image_{context}"]
+= Creating a ModuleLoader image
+
+Kernel Module Management (KMM) works with purpose-built `ModuleLoader` images.
+These are standard OCI images that must satisfy the following requirements:
+
+* `.ko` files must be located under `+/opt/lib/modules/${KERNEL_VERSION}+`
+* `modprobe` and `sleep` binaries must be defined in the `$PATH` variable.
+
+== `depmod`
+
+Run `depmod` at the end of the build process to generate `modules.dep` and `map` files.
+This is best practice if your `ModuleLoader` image contains several kernel modules and if one of the modules depends on
+another module.
+To generate dependencies and map files for a specific kernel version, run `+depmod -b /opt ${KERNEL_VERSION}+`.
+
+== Example `Dockerfile`
+
+The example below builds a test kernel module from the KMM repository.
+Note that you must have a Red Hat subscription to download the `kernel-devel` package.
+If you are building your image on {product-title}, consider using the Driver Tool Kit (DTK).
+
+For further information, see https://cloud.redhat.com/blog/how-to-use-entitled-image-builds-to-build-drivercontainers-with-ubi-on-openshift[using an entitled build].
+
+[,dockerfile]
+----
+FROM registry.redhat.io/ubi8/ubi as builder
+
+ARG KERNEL_VERSION
+
+RUN dnf install -y \
+    gcc \
+    git \
+    kernel-devel-${KERNEL_VERSION} \
+    make
+
+WORKDIR /usr/src
+
+RUN ["git", "clone", "https://github.com/rh-ecosystem-edge/kernel-module-management.git"]
+
+WORKDIR /usr/src/kernel-module-management/ci/kmm-kmod
+
+RUN KERNEL_SRC_DIR=/lib/modules/${KERNEL_VERSION}/build make all
+
+FROM registry.redhat.io/ubi8/ubi-minimal
+
+ARG KERNEL_VERSION
+
+RUN microdnf install kmod
+
+COPY --from=builder /usr/src/kernel-module-management/ci/kmm-kmod/kmm_ci_a.ko /opt/lib/modules/${KERNEL_VERSION}/
+COPY --from=builder /usr/src/kernel-module-management/ci/kmm-kmod/kmm_ci_b.ko /opt/lib/modules/${KERNEL_VERSION}/
+
+RUN depmod -b /opt ${KERNEL_VERSION}
+----
+
+== Building in cluster
+
+KMM is able to build ModuleLoader images in cluster.
+Build instructions must be provided using the `build` section of a kernel mapping.
+The `Dockerfile` for your container image should be copied into a `ConfigMap` object, under the `Dockerfile` key.
```

Comment:
Why not add the full example of the configmap?
In my opinion it will be helpful for customer to have a full example.

Reply:
I will ask Quba for a full configmap.